### DISCUSSION TITLE
many to many One-way？

### DISCUSSION THREAD

**Discussion by xdewx** [sqlalchemy/sqlalchemy#8186] [Usage Questions] [answered]
### Describe the use case

```python
class Tag(Base):
    id=Column(...)
    key=Column(...)
    value=Column(...)

class MixinTag:
    __abstract__=True
    @declared_attr
    def tags(cls):
        return relationship(Tag)

class A(Base,MixinTag):
    id=Column(...)
class B(Base,MixinTag):
    id=Column(...)

class ModelTagMapping(Base):
    model_name=Column(String)
    model_id=Column(Integer,ForeignKey(f"{model_name}.id"))
    tag_id=Column(Integer,ForeignKey("tag.id"))
```

The above code is **pseudocode**, and an error will be reported during actual operation. **How should I fix it**.

I want to achieve the effect that all the tags exists in only one table and the remaining tables refer to multiple pieces of data in the tag

### Databases / Backends / Drivers targeted

sql

### Example Use

```python
tag=Tag(key='test',value='test')
a=A(tags=[tag])
b=B(tags=[tag])
session.add(a)
session.add(b)
session.commit()
```

**Comment by CaselIT** [comment]
Hi,

I'm not sure what "this" is. Could you provide more details?

**Comment by xdewx** [comment]
> Hi,
> 
> I'm not sure what "this" is. Could you provide more details?

sorry. i modified my question. please help me.

**Comment by CaselIT** [comment, marked as answer]
Thanks for updating,

Take a look at these examples https://docs.sqlalchemy.org/en/14/orm/examples.html#module-examples.generic_associations that contain some similar solutions
